Landing page and form template

danielcb29 opened the pull request:
Hey Guys! 

This PR has the angular version of the template selected on #5 😀 

The landing page has 4 components: 
- Header
- Content
- Footer
- Contat 

The form is other component and is inside contact component 🌵 

Plase review the code and the result, I suggest to pull the branch and take a deep look serving the application. 

We need help ✋ to define the content (information and pics) of landing page. 

Next version will include all content and form functionality 😎 

Some screenshots: 

![agm1](https://user-images.githubusercontent.com/8260962/27750576-024b5db6-5d9e-11e7-988b-c4b8d131989b.PNG)

![agm2](https://user-images.githubusercontent.com/8260962/27750577-040b7a14-5d9e-11e7-92fd-8d2804ec296a.PNG)

![agm3](https://user-images.githubusercontent.com/8260962/27750581-093438c8-5d9e-11e7-9e9b-b56faa498264.PNG)

![agm4](https://user-images.githubusercontent.com/8260962/27750588-0ed142ee-5d9e-11e7-8ed2-868295badf44.PNG)

![agm5](https://user-images.githubusercontent.com/8260962/27750591-12f1c772-5d9e-11e7-9877-b865813ae998.PNG)

alejandronanez commented on src/app/contact/contact.component.spec.ts:
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
General question: Are these generated files by `angular-cli`? If so and if we're not doing unit tests for this iteration I think we can delete/skip any `*.spec.ts` file.

jdjuan replied:
We can skip unit testing for now! No need to delete them though! 😁

danielcb29 replied:
Sure, we can ignore the .spec files, however, this tests are generated by ```angular-cli``` and checks the component bootstrapping. 😄

alejandronanez commented on src/app/footer/footer.component.ts:
```diff
@@ -0,0 +1,15 @@
+
+  constructor() { }
+
+  ngOnInit() {
```
I'm pretty sure `onInit` and `constructor` are not needed since we're not calling anything inside. You can try delete them in this component and in any other place that we're not using them.

jdjuan replied:
Yeah, let's remove `ngOnInit()` but let's leave the constructor, it is created by default, it does no harm and it is more likely to be used at some point than not :wink:

alejandronanez commented on src/app/form/form.component.css:
```diff
@@ -0,0 +1,151 @@
+	background:#c8c8c8;
+	font:600 16px/18px 'Open Sans',sans-serif;
+}
+*,:after,:before{box-sizing:border-box}
```
Do you mind instead of
```css
*,:after,:before{box-sizing:border-box}

// we do
*, :after, :before {
    box-sizing:border-box
}
```
That's more readable / easier to diff later one than a one liner.

jdjuan replied:
Right right right! this guy @alejandronanez appreciates readable beautiful code and I agree with him! Its advice is really good! 😉

alvaromartinez986 replied:
Sorry about that, this css is not being used, so I am going to delete it. :yum:

danielcb29 replied:
It shouldn't be there because it is the css from old form version. Sorry 😅

alejandronanez commented on src/app/form/form.component.css:
```diff
@@ -0,0 +1,151 @@
+body{
```
We should add a css linter that checks for what we're using for spacing. (spaces vs tabs) I don't have a strong opinion tbh but this code looks like it has like 6 spaces per rule and that's kinda weird.

jdjuan replied:
Right, let's not forget to use `Shift + Alt + f` to have VSCode editor format the code. Angular comes with a linter by default, let's just run `ng lint` before pushing! 👍

alejandronanez commented on src/app/header/header.component.css:
```diff
@@ -0,0 +1,5 @@
+#banner {
+    background-image: url("../../assets/img/angular-medellin-big.png");
```
Indentation here looks fine 👍

alejandronanez commented:
I suggest we should start using either `npm@5` or `yarn` to ensure deterministic installs. We can do this in another PR though.

alejandronanez commented:
Also, great work!!!

jdjuan commented:
@danielcb29 @alvaromartinez986 Great job guys! Keep it up! Thanks for you efforts! Don't mind toooo much the design, we'll have a design make it look awesome at some point 😉 

Fix those comments and we'll approve

PS: @alejandronanez yeah, `yarn` and I have been dating lately and we love each other. Yarn will be more likely the default soon!

danielcb29 commented:
@alejandronanez @jdjuan We made the corrections, thank you guys!!, I think we can create a git pre_hook to run ```ng lint``` and ```ng test```before push. 😄 , just waiting for new comments or +1

danielcb29 commented:
to add the pre-push hook follow the next steps: 

1) Rename the file in .git/hooks/  pre-push.sample to pre-push
   ```mv pre-push.sample pre-push```

2) Add the next code to pre-push file 
    ```
      # Run ng ling and ng test before pushing
      echo 'Running ng lint and ng test before your push...'
      ng lint | ng test
    ```

3) Enjoy your new hook 🍦 

I didn't test it but it should work, any suggestion or enhancement please let me know

alejandronanez commented on src/app/footer/footer.component.ts:
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core';
```
You can get rid of the OnInit import, that's dead code.

jdjuan replied:
Right! That's correct!

alejandronanez commented on .angular-cli.json:
```diff
@@ -19,7 +19,9 @@
       "prefix": "app",
       "styles": [
-        "styles.css"
+        "styles.css",
```
I think we whould add styles.css in the last position here. We don't want to any .css overwrites our style.css.

```json
[
    "../node_modules/....",
    "../node_modules/....",
    "styles.css"
]
```

jdjuan replied:
Makes sense! 👍

jdjuan commented:
@danielcb29 and @alvaromartinez986 the prehook thing is awesome! Kudos to you for that! 🥇  I didn't really thought of it before!

danielcb29 commented:
@alejandronanez @jdjuan Thanks for comments! , the PR is updated now. The hook configuration are local, so, I updated the comment of hook implementation and I changed my local hook to run only ```ng lint```.  Please take a look of new changes and let me know any new change you think would be needed. 💡

alejandronanez commented:
@danielcb29 If you want to create git hooks that can be controlled using npm you can try out this package https://github.com/typicode/husky I think that's a better approach than copying configurations.

danielcb29 commented:
@alejandronanez Thats awesome!, let me update the project to use husky
